### ISSUE TITLE
Unexpected 'found unparsable section' errors.

### ISSUE THREAD

**Issue by nevdelap** [sqlfluff/sqlfluff#4793]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Unexpected 'found unparsable section' errors.

### Expected Behaviour

While this passes...
```
USE ROLE ACCOUNTADMIN;
CREATE WAREHOUSE TEMP_TESTING
WAREHOUSE_SIZE = XSMALL
AUTO_SUSPEND = 60
COMMENT = 'sdfsdfdsf';
DROP WAREHOUSE TEMP_TESTING;   -- It doesn't complain about this.
```

### Observed Behaviour

this is the minimal example that fails with sqlfluff, version 1.4.5.
```
USE ROLE ACCOUNTADMIN;
CREATE WAREHOUSE TEMP_TESTING
WAREHOUSE_SIZE = XSMALL
COMMENT = 'sdfsdfdsf'
AUTO_SUSPEND = 60;
DROP WAREHOUSE TEMP_TESTING;  -- It shouldn't complain about this.
```
which produces...
```
== [sql/temp.sql] FAIL
L:   4 | P:  22 |  PRS | Line 4, Position 22: Found unparsable section:
                       | '\nAUTO_SUSPEND = 60'
```

### How to reproduce

Is in Observed Behaviour.

### Dialect

snowflake

### Version

1.4.5

### Configuration

```
[sqlfluff]
# See https://docs.sqlfluff.com/en/stable/rules.html
exclude_rules = L032, L034, L051
large_file_skip_byte_limit = 50000

[sqlfluff:layout:type:comma]
line_position = leading
spacing_before = touch

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true

[sqlfluff:templater:dbt]
project_dir = /sql
profiles_dir = /.dbt
profile = default
target = dev

[sqlfluff:indentation]
indented_joins = True
template_blocks_indent = True

[sqlfluff:rules]
indent_unit = space
max_line_length = 120
tab_space_size = 4

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L024]

[sqlfluff:rules:L030]
extended_capitalisation_policy = upper

[sqlfluff:rules:L040]
capitalisation_policy = upper

[sqlfluff:rules:L063]
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by nevdelap** [comment]
Thank you!
